src/main/java/de/jtheb/fh/lagerverwaltung/entities/Shelf.java

```diff
@@ -0,0 +1,66 @@
+package de.jtheb.fh.lagerverwaltung.entities;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Shelf {
+    public static final int COMPARTMENTCOUNT = 100;
+
+    private List<Compartment> compartments = new ArrayList<>();
+
+    public Shelf() {
+        this.compartments = new ArrayList<>();
+    }
+
+    public boolean itemFits(Item item) {
+        if (!isFull()) {
+            for (Compartment compartment : compartments) {
+                if (compartment.itemFitsWithArticleNr(item) || compartment.getArticleNr().equals(null)) {
```

Comment:
did you test the latter part? it will never be true, as something.equal(null) would return throw a NullPointerException if something were true